GVFS/GVFS.Service/ProductUpgradeTimer.cs

```diff
@@ -0,0 +1,81 @@
+﻿using GVFS.Common;
+using GVFS.Common.Tracing;
+using GVFS.Upgrader;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace GVFS.Service
+{
+    public class ProductUpgradeTimer
+    {
+        private static readonly TimeSpan TimeInterval = TimeSpan.FromDays(1);
+        private JsonTracer tracer;
+        private Timer timer;
+
+        public ProductUpgradeTimer(JsonTracer tracer)
+        {
+            this.tracer = tracer;
+        }
+
+        public void Start()
+        {
+            Random random = new Random();
+            TimeSpan startTime = TimeSpan.FromMinutes(random.Next(0, 60));

```

Comment:
If `startTime` ends up being 0, `TimerCallback` will be called immediately, is that okay?

Reply:
Yes. Have tested it.